### DISCUSSION TITLE
Real-life use case for serverless services

### DISCUSSION THREAD

**Discussion by ahmedraed101** [HossamAbubakr/Udacity-Full-Stack-Classroom#5] [Q&A] [answered]
Can you please give me a scenario that would be best to use AWS-lambda or any other serverless services

**Comment by HossamAbubakr** [comment, marked as answer]
Hello Ahmed,

That's a great question! As we discussed in the session, serverless functions are best suited to high latency (Due to delay occurring due to a cold boot) and low to medium load functionality.

A chatbot is one of the most common usages for a serverless service. 
For the following reasons:
1. The initial delay will be mostly tolerated.
2. Once the initial request is made, the service will remain "hot" for the duration of the interaction.
3. It's a relatively low to medium demand functionality.

I would also like to share with you these fantastic [case studies](https://serverlessfirst.com/real-world-serverless-case-studies/) on the usage of serverless services.
